### DISCUSSION TITLE
how to select somepart of text in group by

### DISCUSSION THREAD

**Discussion by MHR00** [datalust/seq-tickets#2023] [Help] [answered]
we have text like this : 
header : Content-Length:32, AID:something, OS:Web , .....
and I want 
select * from stream 
group by something

I want use SUBSTRING with CHARINDEX but CHARINDEX seems not support in seq

**Comment by nblumhardt** [comment]
Hi! If `something` is fixed-width, you can use `substring(header, indexof(header, 'AID:') + 4, N)`, where `N` is the length of `something` to pull out. Needs some more thought, otherwise.

**Comment by MHR00** [comment]
`something ` is not fixed-width . 

is there any solution?

**Reply by nblumhardt** [reply, marked as answer]
Is there something that reliably marks the end of `something` in the string, even if it's a part of `something` itself? If there's always `, OS: Web` immediately afterwards, you can use:

```sql
let |start: indexof(header, 'AID:') + 4| substring(header, start, indexof(header, ', OS:Web') - start)
```

Otherwise no, I don't think so, right now. Logging the headers as a (structured) dictionary would be the way to go.

Hope this helps,
Nick
